atlasdb-coordination-impl/src/main/java/com/palantir/atlasdb/coordination/CoordinationServiceImpl.java

```diff
@@ -48,6 +49,11 @@ public CoordinationServiceImpl(CoordinationStore<T> store) {
         return Optional.of(cachedReference);
     }
 
+    /**
+     * In case of failure, the returned value and bound are guaranteed to be the ones that were in the KVS at the time
+     * of CAS failure only if {@link KeyValueService#getCheckAndSetCompatibility()} is
+     * {@link com.palantir.atlasdb.keyvalue.api.CheckAndSetCompatibility#SUPPORTED_DETAIL_ON_FAILURE}.
+     */
```

Comment:
I think this one should be along the lines of "in case of failure, the returned value and bound are guaranteed to be the ones that caused the CAS to fail only if the implementation of the CoordinationStore makes such guarantees".

We shouldn't mention `KeyValueService` at this layer, since I could always write a `TimeLockPaxosCoordinationStore`, for example.